activation/nipost.go

```diff
@@ -322,15 +322,15 @@ func (nb *NIPostBuilder) BuildNIPost(ctx context.Context, challenge *types.NIPos
 }
 
 // Submit the challenge to a single PoET.
-func (nb *NIPostBuilder) submitPoetChallenge(ctx context.Context, poet PoetProvingServiceClient, prefix, challenge []byte, signature types.EdSignature, nodeID types.NodeID) (*types.PoetRequest, error) {
-	poetServiceID, err := poet.PoetServiceID(ctx)
+func (nb *NIPostBuilder) submitPoetChallenge(ctx context.Context, client PoetProvingServiceClient, prefix, challenge []byte, signature types.EdSignature, nodeID types.NodeID) (*types.PoetRequest, error) {
+	poetServiceID, err := client.PoetServiceID(ctx)
 	if err != nil {
 		return nil, &PoetSvcUnstableError{msg: "failed to get PoET service ID", source: err}
 	}
 	logger := nb.log.WithContext(ctx).WithFields(log.String("poet_id", hex.EncodeToString(poetServiceID.ServiceID)))
 
 	logger.Debug("querying for poet pow parameters")
-	powParams, err := poet.PowParams(ctx)
+	powParams, err := client.PowParams(ctx)
```

Comment:
Why not set a timeout here as well? Perhaps it's best to set timeout for entire `submitPoetChallenge()` instead?

Btw, there is a timeout already set for `submitPoetChallenges()` - till the start of the round, which seems reasonable. Are timeouts per poet client necessary?

Reply:
OK, that's true bit it will still hang until the ctx is canceled (at PoET round start).

Reply:
Yes. 🤔 and this is bad because if other submits failed too it won't have time to retry...